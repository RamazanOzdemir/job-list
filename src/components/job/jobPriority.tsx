import { useJobContext } from "@/hooks";
import { StyledPriorityBox } from "@/styles";
import { FC, useMemo } from "react";

type JobPriorityProps = {
  priority: string;
};

const priorityColors = {
  "1": "error",
  "2": "warning",
  "3": "success",
};

export const JobPriority: FC<JobPriorityProps> = ({ priority }) => {
  const { priorities } = useJobContext();

  const priorityMap: Map<string, string> = useMemo(() => {
    const mapList = new Map<string, string>();
    priorities.forEach((priority) => mapList.set(priority.id, priority.name));
    return mapList;
  }, [priorities.length]);

  return (
    <StyledPriorityBox variant={priorityColors[priority] || "error"}>
      {priorityMap.get(priority)}
    </StyledPriorityBox>
  );
};
