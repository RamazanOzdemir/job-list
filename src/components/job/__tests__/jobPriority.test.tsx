import renderer from "react-test-renderer";
import { JobPriority } from "../jobPriority";
import { render } from "@testing-library/react";

describe("Job Priority tests", () => {
  test("expect snapshot test", () => {
    const component = renderer.create(<JobPriority priority="1" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("expect variant is error  when priority is 1", () => {
    const { container } = render(<JobPriority priority="1" />);
    expect(container.querySelector("[variant='error']")).not.toBeNull();
  });

  test("expect variant is warning when priority is 2", () => {
    const { container } = render(<JobPriority priority="2" />);
    expect(container.querySelector("[variant='warning']")).not.toBeNull();
  });

  test("expect variant is success when priority is 3", () => {
    const { container } = render(<JobPriority priority="3" />);
    expect(container.querySelector("[variant='success']")).not.toBeNull();
  });
});
