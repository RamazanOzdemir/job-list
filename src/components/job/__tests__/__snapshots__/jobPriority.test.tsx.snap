// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Priority tests except snapshot test 1`] = `
<div
  className="MuiBox-root css-1xo7tt7"
  variant="error"
/>
`;

exports[`Job Priority tests expect snapshot test 1`] = `
<div
  className="MuiBox-root css-1xo7tt7"
  variant="error"
/>
`;
